### ISSUE TITLE
How do I change the default template?

### ISSUE THREAD

**Issue by ArtemisFowl-san** [lervag/apy#71]
Hey, thanks for this wonderful program, I had the same wish to not have anki open all the time while adding cards.

I am wondering thought when I use apy add I get a sort of premade markdown template of a card. Is there any way to change what that looks like. I tried with the presets, but could not really figure out how to change the default deck for example.

**Comment by lervag** [comment]
Hi! I'm glad you find `apy` useful!

> I am wondering thought when I use apy add I get a sort of premade markdown template of a card. Is there any way to change what that looks like.

I'm not quite sure exactly what you want here. Could you show me an example? I.e., a before and after image or example file contents?

**Comment by ArtemisFowl-san** [comment]
<img width="516" alt="Screenshot 2023-09-12 at 13 28 11" src="https://github.com/lervag/apy/assets/72747914/cf215213-b7a8-441b-a99f-8f4183cf853a">

So, when you use apy add this is the layout you typically get. I do not really use tags and as far as I can tell the "Note" part is also not relevant usually. Is there a way to get rid of these from the template so that it is a bit simplified? I know I can use apy add-single for simple cards, but I generally do prefer entering via "apy add", because it gives me more structure.

**Comment by lervag** [comment]
The `Note` part is necessary because this command relies on the Markdown "translation" of an Anki note. The idea is that you need to put your question inside the `Front` section and the answer in the `Back` section.

The `apy add-single` really does seem relevant to you here.

But perhaps what you want is a simpler format for specifying a card that still uses an editor? I'm not sure if I want to implement it, because there are a lot of gotchas. E.g., we need a structure that represents all the fields you need. "Front" and "Back" for the "Basic" note type, but this may be very different for other note types.

To be honest, it is still a little bit unclear what you want. Could you try to explain it again slightly more verbosely?

**Comment by ArtemisFowl-san** [comment]
I basically wanted the flashcard ordered more as shown in the picture. 

<img width="494" alt="Screenshot 2023-10-13 at 21 49 33" src="https://github.com/lervag/apy/assets/72747914/872c776f-70f2-41b3-af31-eac2d308b921">

> The Note part is necessary because this command relies on the Markdown "translation" of an Anki note. The idea is that you need to put your question inside the Front section and the answer in the Back section.

I think this was the part I was missing. So the reason why the note is structured the way it is with Model at the top and then tags is so that it can be translated from Markdown to Anki? I thought it was annoying that the first line was not the front of the flashcard, so that I could start typing right away and then worry about tags and deck after having written the flashcard. But if this order is required for the translation I understand why it has to be this way. 

I hope I am communicating this better now :)

**Comment by lervag** [comment]
> I hope I am communicating this better now :)

Yes, thanks!

> > The Note part is necessary because this command relies on the Markdown "translation" of an Anki note. The idea is that you need to put your question inside the Front section and the answer in the Back section.
> 
> I think this was the part I was missing. So the reason why the note is structured the way it is with Model at the top and then tags is so that it can be translated from Markdown to Anki?

Yes, precisely. We need to convert between pure text and the data format that Anki uses to represent a note. A note can have multiple fields (e.g. Back and Front for the Basic note type, but perhaps Country, Capitol, and Area for a Geography note type).

> I thought it was annoying that the first line was not the front of the flashcard, so that I could start typing right away and then worry about tags and deck after having written the flashcard. But if this order is required for the translation I understand why it has to be this way.

I think the current idea of using a Markdown document where the level 2 headers indicate the fields is good. Of course, there are other possible ways to do it, but it needs to be well thought out. Your proposal is not good, IMHO, because it is hard to parse. We would need to put the note and tags at the top, but they could be optional. And we need some way to distinguish the fields.

I'm not sure I want to actually add support for another format, though. I'm not sure it really provides any significant improvement.

----

One thought: If you would indicate your current workflow of using apy to add new cards, then I might be able to provide some suggestions on how to improve that.

**Comment by lervag** [comment]
Closed due to inactivity.
